versioning/lockfiles.rst

```diff
@@ -130,47 +130,141 @@ For more information see :ref:`commands`
 
-    (EXAMPLE COMING SOON!)
+    $ conan graph build-order ./release --json=bo.json --build=missing
+    [[PkgC, PkgD], [App]]  # simplified format
+
+This command will return a list of lists, in order, of those packages to be built. It will be
+stored in a *bo.json* json file too. Note that the ``--build=missing`` follows the same rules
+as :command:`create` and :command:`install` commands. The result of evaluating the graph with
+the **PkgA/0.2** version, due to the ``full_version_mode`` policy is that new binaries for
+PkgB, PkgC and App are necessary, and they do not exist yet. If we don't provide the ``--build=missing``
+it will return an empty list (but it will fail later, because binary packages are not available).
+
+We can now proceed iteratively with the following procedure:
+
+
+1. pop the first element of the first sublist of the build order result, get its ``ref`` reference
+
+    .. code:: python
+
+        # python
+        _, ref = build_order[0][0]
+        ref = ref.split("#", 1)[0]
+
+2. allocate some resource, like a CI build server, or create a temporary folder.
+
+    .. code:: bash
+
+        $ mkdir build_server_folder && mkdir build_server_folder/release
+
+3. copy the lockfile to that resource (and move to it)
+
+    .. code:: bash
+
+        $ cp release/conan.lock build_server_folder/release
+        $ cd build_server_folder
+
+4. clean "modified" nodes from the lockfile
+
+    .. code:: bash
+
+        $ conan graph clean-modified release/
+
+5. build the package
+
+    .. code:: bash
+
+        $ conan install <ref> --build=<ref> --lockfile=release
+
+6. go back to the parent, update the lockfile with the changes
+
+    .. code:: bash
+
+        $ cd ..
+        $ conan graph update-lock release build_server_folder/release
+        $ rm -rf build_server_folder
+
+7. compute again the build-order of packages, if not empty, goto 1
+
+    .. code-block:: bash
+
+        $ conan graph build-order ./release --json=bo.json --build=missing
+
+Note that this is a suboptimal approach, in order to explain the functionality, which
+is more easy to follow if it is sequential. In reality, the CI can take the first
+sublist output of :command:`conan graph build-order` and fire all its packages in parallel,
+because they are guaranteed to be independent. Then, as soon as they start finishing and
+build servers become available, the :command:`conan graph build-order` can be reevaluated,
+and new builds can be launched accordingly, just taking care of not re-launching the same
+build again. Note that the result of build-order contains a unique UUID, which is the identifier
+of the node in the graph, which could be useful to dissambiguate.
 
+.. image:: ../images/lockfile_ci_2.png
 
-.. image:: ../images/lockfile_ci_2.png
\ No newline at end of file
+With this later approach, a deterministic build with optimal Continuous Integration process
+with optimal utilization of resources and minimizing unnecessary rebuilds is achieved.
```

Comment:
I think we have to write a paragraph with the approach using revisions, without constant version bumping, without using necessarily version ranges and adjusting the default mode to package_revision_mode.

Reply:
Maybe also another paragraph clarifying to it could be done in a real CI, using binary storage like Artifactory to store and retrieve the lock files for the build (generic repository).